src/unused_finder/unused_finder.rs

```diff
@@ -0,0 +1,323 @@
+use std::{
+    collections::{BTreeMap, HashMap, HashSet},
+    iter::FromIterator,
+    str::FromStr,
+    sync::Arc,
+};
+
+use napi_derive::napi;
+use rayon::prelude::*;
+use swc_core::{
+    common::source_map::Pos,
+    ecma::loader::resolvers::{
+        lru::CachingResolver, node::NodeModulesResolver, tsc::TsConfigResolver,
+    },
+};
+
+use crate::import_resolver::TsconfigPathsJson;
+
+use super::{
+    create_caching_resolver, create_flattened_walked_files, create_report_map_from_flattened_files,
+    graph::{Graph, GraphFile},
+    process_import_export_info, read_allow_list,
+    unused_finder_visitor_runner::get_import_export_paths_map,
+    ExportedItemReport, FindUnusedItemsConfig, UnusedFinderReport, WalkFileMetaData,
+};
+
+#[derive(Debug, Clone, Default)]
+#[napi(js_name = "UnusedFinder")]
+pub struct UnusedFinder {
+    pub report: UnusedFinderReport,
+    pub logs: Vec<String>,
+    config: FindUnusedItemsConfig,
+    entry_packages: HashSet<String>,
+    file_path_exported_items_map: HashMap<String, Vec<ExportedItemReport>>,
+    skipped_items: Arc<Vec<regex::Regex>>,
+    skipped_dirs: Arc<Vec<glob::Pattern>>,
+    tsconfig: TsconfigPathsJson,
+    flattened_walk_file_data: Vec<WalkFileMetaData>,
+    graph: Graph,
+    // resolver: CachingResolver<TsConfigResolver<NodeModulesResolver>>,
```

Comment:
I'm assuming this was intended to be cached, but we are discarding the resolution cache on every evaluation because.. the resolutions might change with file edits?

This means we're not actually caching resolutions